[Bug]: Provider crash when reading state of an ECR repository

### Terraform Core Version

1.4.0,1.4.1

### AWS Provider Version

4.56.0,4.58.0

### Affected Resource(s)

* aws_ecr_repository

### Expected Behavior

Provider either succeeds in getting the state of the registry, or return an error that can help me understand what went wrong

### Actual Behavior

Provider crashed

### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v4.56.0_x5 plugin:

panic: interface conversion: interface {} is nil, not *ecr.Repository

goroutine 398 [running]:

github.com/hashicorp/terraform-provider-aws/internal/service/ecr.resourceRepositoryRead({0xe5376c0?, 0xc0033be720}, 0xc00348fb00, {0xd091680?, 0xc0003fb400})

github.com/hashicorp/terraform-provider-aws/internal/service/ecr/repository.go:185 +0x12cc

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xe5376c0?, {0xe5376c0?, 0xc0033be720?}, 0xd?, {0xd091680?, 0xc0003fb400?})

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:719 +0x87

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00119d420, {0xe5376c0, 0xc0033be720}, 0xc0033691e0, {0xd091680, 0xc0003fb400})

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:1015 +0x585

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000011548, {0xe5376c0?, 0xc0033be600?}, 0xc003382200)

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:613 +0x4a5

github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0xc0033597a0, 0xc003359800, {0xc004905620, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)

github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ReadResource.go:26 +0x102

github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc001ec97c0, {0xe5376c0?, 0xc0033b4300?}, 0xc0034a0900)

github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1

github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xcda6b00?, 0xc001ec97c0}, {0xe5376c0, 0xc0033b4300}, 0xc004aaecb0, 0x0)

github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170

google.golang.org/grpc.(*Server).processUnaryRPC(0xc00003f2c0, {0xe546640, 0xc004a104e0}, 0xc0034d6480, 0xc004a05200, 0x14c30450, 0x0)

google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23

google.golang.org/grpc.(*Server).handleStream(0xc00003f2c0, {0xe546640, 0xc004a104e0}, 0xc0034d6480, 0x0)

google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f

google.golang.org/grpc.(*Server).serveStreams.func1.2()

google.golang.org/grpc@v1.53.0/server.go:965 +0x98

created by google.golang.org/grpc.(*Server).serveStreams.func1

google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-aws_v4.56.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely

helpful if you could report the crash with the plugin's maintainers so that it

can be fixed. The output above should help diagnose the issue.

{"@level":"info","@message":"data.doormat_aws_credentials.packer_team_dev: Refreshing...","@module":"terraform.ui","@timestamp":"2023-03-16T15:23:22.918752Z","hook":{"resource":{"addr":"data.doormat_aws_credentials.packer_team_dev","module":"","resource":"data.doormat_aws_credentials.packer_team_dev","implied_provider":"doormat","resource_type":"doormat_aws_credentials","resource_name":"packer_team_dev","resource_key":null},"action":"read"},"type":"apply_start"}

Stack trace from the terraform-provider-aws_v4.56.0_x5 plugin:

panic: interface conversion: interface {} is nil, not *ecr.Repository

goroutine 398 [running]:

github.com/hashicorp/terraform-provider-aws/internal/service/ecr.resourceRepositoryRead({0xe5376c0?, 0xc0033be720}, 0xc00348fb00, {0xd091680?, 0xc0003fb400})

github.com/hashicorp/terraform-provider-aws/internal/service/ecr/repository.go:185 +0x12cc

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xe5376c0?, {0xe5376c0?, 0xc0033be720?}, 0xd?, {0xd091680?, 0xc0003fb400?})

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:719 +0x87

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00119d420, {0xe5376c0, 0xc0033be720}, 0xc0033691e0, {0xd091680, 0xc0003fb400})

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:1015 +0x585

github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000011548, {0xe5376c0?, 0xc0033be600?}, 0xc003382200)

github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:613 +0x4a5

github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0xc0033597a0, 0xc003359800, {0xc004905620, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)

github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ReadResource.go:26 +0x102

github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc001ec97c0, {0xe5376c0?, 0xc0033b4300?}, 0xc0034a0900)

github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1

github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xcda6b00?, 0xc001ec97c0}, {0xe5376c0, 0xc0033b4300}, 0xc004aaecb0, 0x0)

github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170

google.golang.org/grpc.(*Server).processUnaryRPC(0xc00003f2c0, {0xe546640, 0xc004a104e0}, 0xc0034d6480, 0xc004a05200, 0x14c30450, 0x0)

google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23

google.golang.org/grpc.(*Server).handleStream(0xc00003f2c0, {0xe546640, 0xc004a104e0}, 0xc0034d6480, 0x0)

google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f

google.golang.org/grpc.(*Server).serveStreams.func1.2()

google.golang.org/grpc@v1.53.0/server.go:965 +0x98

created by google.golang.org/grpc.(*Server).serveStreams.func1

google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-aws_v4.56.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely

helpful if you could report the crash with the plugin's maintainers so that it

can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "= 4.56.0"
    }

    doormat = {
      source  = "<internal>"
      version = "~> 0.0.2"
    }
  }

  backend "remote" {
    hostname     = "app.terraform.io"
    organization = "<org>"

    workspaces {
      name = "<workspace>"
    }
  }
}

data "doormat_aws_credentials" "packer_team_dev" {
  provider = doormat

  role_arn = var.doormat_role_arn
}

provider "aws" {
  region = var.region

  access_key = data.doormat_aws_credentials.packer_team_dev.access_key
  secret_key = data.doormat_aws_credentials.packer_team_dev.secret_key
  token      = data.doormat_aws_credentials.packer_team_dev.token
}

resource "aws_ecr_repository" "<tele>" {
  name                 = "<repo>"
  image_tag_mutability = "IMMUTABLE"
  force_delete         = false
}
```

### Steps to Reproduce

Local plan/apply works without an error, the panic only occurs on TFC. I suspect the authentication does not occur normally, but the doormat provider does not error. I may have some outstanding errors in my configuration either in the terraform files, or in the TFC variables/AWS IAM.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

Hashicorp internal environment

### References

_No response_

### Would you like to implement a fix?

Yes

--- a/.changelog/30067.txt
+++ b/.changelog/30067.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ecr_repository: Fix unhandled errors and nil output on read
+```

--- a/internal/service/ecr/repository.go
+++ b/internal/service/ecr/repository.go
@@ -2,6 +2,7 @@ package ecr
 
 import (
 	"context"
+	"errors"
 	"log"
 	"time"
 
@@ -14,10 +15,12 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
 	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
 // @SDKResource("aws_ecr_repository")
@@ -109,6 +112,10 @@ func ResourceRepository() *schema.Resource {
 	}
 }
 
+const (
+	ResNameRepository = "Repository"
+)
+
 func resourceRepositoryCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).ECRConn()
@@ -181,6 +188,12 @@ func resourceRepositoryRead(ctx context.Context, d *schema.ResourceData, meta in
 		log.Printf("[WARN] ECR Repository (%s) not found, removing from state", d.Id())
 		d.SetId("")
 		return diags
+	}
+	if err != nil {
+		return create.DiagError(names.ECR, create.ErrActionReading, ResNameRepository, d.Id(), err)
+	}
+	if outputRaw == nil {
+		return create.DiagError(names.ECR, create.ErrActionReading, ResNameRepository, d.Id(), errors.New("empty output"))
 	}
 
 	repository := outputRaw.(*ecr.Repository)

